src/main/java/uk/gov/pay/adminusers/model/AssignServiceAndRoleRequest.java

```diff
@@ -0,0 +1,24 @@
+package uk.gov.pay.adminusers.model;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import io.dropwizard.validation.ValidationMethod;
+
+import javax.validation.constraints.NotBlank;
+import javax.ws.rs.WebApplicationException;
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public record AssignServiceAndRoleRequest(
+        @NotBlank(message = "Field [service_external_id] is required") String serviceExternalId,
+        String roleName) {
```

Comment:
I'm not sure why this can't just have type `RoleName` with `@Valid` annotation so you wouldn't need the `isValidRoleName` method but no doubt there are reasons.

Reply:
I originally went with using a custom deserialiser to map the string into a `RoleName` but Jersey returns a non-422 status code. Also with the custom deserialiser we'd need a custom exception class for the response body to contain `"Field [role_name] must be one of 'admin', 'view-and-refund'...."` and it's quite nice we can define that message in this class.